packages/material-ui/src/styles/adaptV4Theme.js

```diff
@@ -74,12 +74,20 @@ export default function adaptV4Theme(inputTheme) {
     ...mixins,
   };
 
+  const { type: paletteMode, ...paletteRest } = palette;
+
   // theme.palette.text.hint
   theme.palette = {
     text: {
-      hint: palette.type === 'dark' ? 'rgba(255, 255, 255, 0.5)' : 'rgba(0, 0, 0, 0.38)',
+      hint:
+        palette.mode === 'dark' || palette.type === 'dark'
+          ? 'rgba(255, 255, 255, 0.5)'
+          : 'rgba(0, 0, 0, 0.38)',
     },
-    ...palette,
+    ...(paletteMode && {
```

Comment:
Am I capturing everything necessary here?

Reply:
Would using `const { type: mode, ...paletteRest } = palette;` simplify this a bit, or am I missing something?

Reply:
You are correct, I over-complicated here :)) let me refactor it